spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/util/TypeUtil.java

```diff
@@ -98,11 +100,33 @@ private static String getSpecialTypeName(String canonicalName) {
     }
 
     public static boolean isPrimitive(Class<?> javaBaseClass, String javaFieldName) {
+        Field field = findField(javaBaseClass, javaFieldName);
+        return field != null && field.getType().isPrimitive();
+    }
+
+    public static Field findField(Class<?> javaBaseClass, String javaFieldName) {
         Field field = ReflectionUtils.findField(javaBaseClass, javaFieldName);
         if (field == null && isGetter(javaFieldName)) {
             field = ReflectionUtils.findField(javaBaseClass, fromGetter(javaFieldName));
         }
-        return field != null && field.getType().isPrimitive();
+        return field;
+    }
+
+    public static Method findMethod(Class<?> javaBaseClass, String javaMethodName) {
+        return ReflectionUtils.findMethod(javaBaseClass, javaMethodName);
+    }
+
+    public static <T extends Annotation> T resolveAnnotation(Class<?> javaBaseClass,
+            String javaFieldName, Class<T> annotationClass) {
+        Field field = findField(javaBaseClass, javaFieldName);
+        if (field != null) {
+            return field.getAnnotation(annotationClass);
+        }
+        Method method = findMethod(javaBaseClass, javaFieldName);
+        if (method != null) {
+            return method.getAnnotation(annotationClass);
+        }
```

Comment:
If there is a method and a field with an annotation, only the annotation from the field is used. Shouldn't it be the other way around and not only take it to account either one, i.e. prefer the annotation on the method over the annotation on the field and fall back to the field if there is no annotation on the method?

Test case:
```
        private int index2;

        @Deprecated
        public int getIndex5() {
            return index;
        }
```
The current implementation would yield that index5 is not deprecated.